### PR TITLE
Release Dafny 3.6.0

### PR DESCRIPTION
**Pull request by cpitclaudel**
Using this PR to run CI; will need another commit to fix release notes on master.

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Deeps test run is at https://github.com/dafny-lang/dafny/actions/runs/2303523148

**Review by robin-aws — DISMISSED** [review]

File: Source/version.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
 using System.Reflection;
-// Version 3.5.0, year 2018+4 month 3 day 14 
-[assembly: AssemblyVersion("3.5.0.40314")]
-[assembly: AssemblyFileVersion("3.5.0.40314")]
+// Version 3.6.0, year 2018+4 month 5 day 10
```
Comment (robin-aws) [inline Source/version.cs:2]: Just a heads up that we'll need to bump this up a day, since we won't be able to actually tag and release until the 11th :)

File: .github/workflows/check-deep-tests-reusable.yml [hunk 2]
```diff
@@ -19,4 +21,7 @@ jobs:
       with:
         script: |
           const script = require('${{ github.workspace }}/dafny/.github/workflows/check-for-workflow-run.js')
-          console.log(script({github, context, core}, workflowID = 'deep-tests.yml', branch = 'master', sha = '${{ inputs.sha }}'))
+          console.log(script({github, context, core},
+                             {workflow_id: 'deep-tests.yml',
+                              sha: '${{ inputs.sha }}' || null,
```
Comment (robin-aws) [inline .github/workflows/check-deep-tests-reusable.yml:26]: The `|| null` seems redundant to me, since `''` is equally falsey.

File: .github/workflows/check-for-workflow-run.js [hunk 3]
```diff
@@ -1,19 +1,19 @@
 // Check for a successful previous run of a workflow.
 // This could make a decent reusable action itself.
-module.exports = async ({github, context, core}, workflowID, branch, sha = null) => {
+module.exports = async ({github, context, core}, {workflow_id, sha, ...config}) => {
```
Comment (robin-aws) [inline .github/workflows/check-for-workflow-run.js:3]:
```suggestion
module.exports = async ({github, context, core, workflow_id, sha, ...config}) => {
```

Might as well pass everything in one object parameter rather than two, right?

File: .github/workflows/check-deep-tests-reusable.yml (hunk 2 above)
Comment (cpitclaudel) [reply .github/workflows/check-deep-tests-reusable.yml:26]: We don't know whether the library treats them the same, do we?

Comment (cpitclaudel) [reply .github/workflows/check-deep-tests-reusable.yml]: Solved by removing the keys from the object completely when they are not given.

File: Source/version.cs (hunk 1 above)
Comment (cpitclaudel) [reply Source/version.cs]: Done

File: .github/workflows/check-for-workflow-run.js (hunk 3 above)
Comment (cpitclaudel) [reply .github/workflows/check-for-workflow-run.js]: Good point!  Amended existing commit.

File: .github/workflows/check-deep-tests-reusable.yml [hunk 4]
```diff
@@ -19,4 +21,7 @@ jobs:
       with:
         script: |
           const script = require('${{ github.workspace }}/dafny/.github/workflows/check-for-workflow-run.js')
-          console.log(script({github, context, core}, workflowID = 'deep-tests.yml', branch = 'master', sha = '${{ inputs.sha }}'))
+          console.log(script({github, context, core,
+                              workflow_id: 'deep-tests.yml',
+                              ...('${{ inputs.sha }}' ? {sha: '${{ inputs.sha }}'} : {}),
+                              ...('${{ inputs.branch }}' ? {branch: '${{ inputs.branch }}'} : {})}))
```
Comment (robin-aws) [inline .github/workflows/check-deep-tests-reusable.yml:27]: Cute! I suspect this isn't particularly idiomatic Javascript but not worth bikeshedding over. :)

Comment (cpitclaudel) [reply .github/workflows/check-deep-tests-reusable.yml:27]: ? It's completely standard, AFAIK.
You might argue that `('${{ inputs.branch }}' && {branch: '${{ inputs.branch }}'})` also complies with ES6, but it relies on corner cases.

Comment (robin-aws) [reply .github/workflows/check-deep-tests-reusable.yml:27]: I meant it would be more idiomatic to pass the parameters unconditionally and check for empty strings on the callee side as needed.

**Review by robin-aws — APPROVED** [review]
